js/Views/Scatterplot.ts

```diff
@@ -0,0 +1,202 @@
+import * as d3 from "d3";
+import {Selection} from 'd3-selection';
+import {IView} from "./IView";
+import {Year_to_indicators_map, State_indicators, State_id} from "../Data/State_indicators"
+import {Dimensions} from "../Utils/svg-utils";
+
+
+export class Scatterplot implements IView 
+{
+    readonly year_to_indicators: Year_to_indicators_map;
+    readonly container: Selection<any, any, any, any>
+    readonly dropdownWrapper: Selection<any, any, any, any>
+    readonly svg: Selection<any, any, any, any>;
+    readonly svgDims: Dimensions;
+    readonly padding: number;
+    readonly axesGroup: Selection<any, any, any, any>;
+    readonly circleGroup: Selection<any, any, any, any>;
+    readonly indicators: Array<string>
+    readonly transition_time: number
+    activeX: string
+    activeY: string
+    xScale: d3.ScaleLinear<number, number>
+    yScale: d3.ScaleLinear<number, number>
+    curYear: number;
+    cur_year_data: Array<State_indicators>;
+
+    constructor(state_data: Year_to_indicators_map, container: Selection<any, any, any, any>,
+                svgDims: Dimensions, startYear: number = 2017) 
+    {
+        this.curYear = startYear;
+        this.year_to_indicators = state_data
+        this.cur_year_data = this.year_to_indicators[this.curYear];
+        this.container = container;
+        this.dropdownWrapper = container.append('div');
+        this.svg = container.append('svg').attr('height', svgDims.height).attr('width', svgDims.width);
+        this.axesGroup = this.svg.append('g');
+        this.circleGroup = this.svg.append('g');
+        this.indicators = ['population', 'total_left', 'total_came', 'net_immigration_flow', 'GDP_per_capita', 'GDP_percent_change', 'jobs', 'jobs_per_capita', 'personal_income_per_capita', 'personal_disposable_income_per_capita', 'personal_taxes_per_capita'];
+        this.activeX = 'jobs_per_capita';
+        this.activeY = 'net_immigration_flow';
+        this.svgDims = svgDims;
+        this.padding = 100;
+        this.transition_time = 800;
+        
+        this.create_dropdowns();
+        this.create_scales();
+        this.update_plot();
+    };
+
+    static indicator_to_name(indicator)
```

Comment:
This is a global problem I've been having, actually.  Might be good to have a standard way to do this.

Reply:
I like the idea of a function like this, but there's more than one place we're doing it, a common place for that stuff would be good